core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -17,27 +18,45 @@
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
 
+import static com.oath.oak.NativeAllocator.OakNativeMemoryAllocator.INVALID_BLOCK_ID;
+import static com.oath.oak.UnsafeUtils.intsToLong;
+
 public class Chunk<K, V> {
 
     /*-------------- Constants --------------*/
 
-    static final int NONE = 0;    // constant for "no index", etc. MUST BE 0!
+    enum OFFSET {
+        NEXT(0), KEY_POSITION(3), VALUE_STATS(1), VALUE_POSITION(1),
+        VALUE_BLOCK_AND_LENGTH(2), VALUE_BLOCK(2), VALUE_LENGTH(2), KEY_BLOCK_AND_LENGTH(4),
+        KEY_BLOCK(4), KEY_LENGTH(4);
+
+        public final int value;
+
+        OFFSET(int value) {
+            this.value = value;
+        }
+    }
+
+    enum State {
+        INFANT,
+        NORMAL,
+        FROZEN,
+        RELEASED
+    }
 
+    static final int NONE = 0;    // constant for "no index", etc. MUST BE 0!
```

Comment:
What "index" is it referenced to? There can be multiple meanings... (I think this is about next-entry-index...)

Reply:
It refers to a null next-index (it is not my addition, it just moved around).

Reply:
Yes, I saw it is not your addition. Please correct the comment anyway, it is just couple of characters :)

Reply:
Will do!